airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -236,6 +257,8 @@ def _resolve_connection(self) -> dict[str, Any]:
             "deploy_mode": None,
             "spark_binary": self.spark_binary or DEFAULT_SPARK_BINARY,
             "namespace": None,
+            "principal": None,
+            "keytab": None,
```

Comment:
Why not separate them into two different connections? There can be many more than just one keytab and principal in each master

Reply:
I just realized what were you meaning 🤔🤣

will push soon! thanks for the suggestion!

Reply:
Don't you have several different users on the same yarn queue?

Reply:
I meant what about there are several different principles, since you only override the user from the connection if he has not entered a principle, this is correct and it will be possible to use several different principles on the same queue.

So I approve this comment